src/domain/entities/Notification.ts

```diff
@@ -27,11 +30,51 @@ export class Notification extends Struct<NotificationAttrs>() {
             });
         }
     }
-    isRead(user: User): boolean {
+
+    isReadBy(user: User): boolean {
         return this.readBy.some(read => read.id === user.id);
     }
-}
 
+    canView(user: User) {
+        return this.checkPermissionAccess(user, "r");
+    }
+
+    canEdit(user: User) {
+        return this.checkPermissionAccess(user, "rw");
+    }
+
+    private checkPermissionAccess(user: User, accessType: "r" | "rw"): boolean {
```

Comment:
[nothing to do] I am not a big fan of using the dhis2 access strings in the domain. It's ok to use D2 specifics when they look good, this string is not nice to use :) But I see we already have in the domain publicAccess and access props as strings, so we will use it. But for new projects, i think it's nicer to have more declarative object types for accesses types.